vignettes/irecist.Rmd

```diff
@@ -0,0 +1,914 @@
+the environment. This will be a company specific process. Some of the data
+frames needed may be `ADSL`, `RS` and `TU`. For this vignette we assume that
+`RS` provides the response values `"iCR"`, `"iPR"`, `"iSD"`,
+`"NON-iCR/NON-iUPD"`, `"iUPD"`, `"iCPD"`, and `"NE"`. All examples can be easily
+modified to consider other response values (see [Handling Different Input
+Response Values](#different_resp_vals)).
+
+For example purpose, the SDTM and ADaM datasets (based on CDISC Pilot
+test data)---which are included in `{pharmaversesdtm}` and `{pharmaverseadam}`---are used.
+
+```{r message=FALSE}
+library(admiral)
+library(admiralonco)
+library(dplyr)
+library(pharmaverseadam)
+library(pharmaversesdtm)
+library(lubridate)
+library(stringr)
+data("adsl")
+# iRECIST oncology sdtm data
+data("rs_onco_irecist")
+
+rs <- rs_onco_irecist
+
+rs <- convert_blanks_to_na(rs)
+```
+
+```{r echo=FALSE}
+# select subjects from adsl such that there is one subject without RS data
+rs_subjects <- unique(rs$USUBJID)
+adsl_subjects <- unique(adsl$USUBJID)
+adsl <- filter(
+  adsl,
+  USUBJID %in% union(rs_subjects, setdiff(adsl_subjects, rs_subjects)[1])
+)
+```
+
+At this step, it may be useful to join `ADSL` to your `RS` domain. Only
+the `ADSL` variables used for derivations are selected at this step. The
+rest of the relevant `ADSL` would be added later.
+
+```{r eval=TRUE}
+adsl_vars <- exprs(RANDDT)
+adrs <- derive_vars_merged(
+  rs,
+  dataset_add = adsl,
+  new_vars = adsl_vars,
+  by_vars = exprs(STUDYID, USUBJID)
+)
+```
+
+```{r, eval=TRUE, echo=FALSE}
+dataset_vignette(
+  adrs,
+  display_vars = exprs(USUBJID, RSTESTCD, RSDTC, VISIT, RANDDT),
+  filter = RSTESTCD == "OVRLRESP"
+)
+```
+
+## Pre-processing of Input Records {#input}
+
+The first step involves company-specific pre-processing of records for
+the required input to the downstream parameter functions. Note that this
+could be needed multiple times (e.g. once for investigator and once for
+Independent Review Facility (IRF)/Blinded Independent Central Review
+(BICR) records). It could even involve merging input data from other
+sources besides `RS`, such as `ADTR`.
+
+This step would include any required selection/derivation of `ADT` or applying
+any necessary partial date imputations, updating `AVAL` (e.g. this should be
+ordered from best to worst response), and setting analysis flag `ANL01FL`.
+Common options for `ANL01FL` would be to set null for invalid assessments or
+those occurring after new anti-cancer therapy, or to only flag assessments on or
+after date of first treatment/randomization, or rules to cover the case when a
+patient has multiple observations per visit (e.g. by selecting the worst value).
+Another consideration could be extra potential protocol-specific sources of
+Progressive Disease such as radiological assessments, which could be
+pre-processed here to create a PD record to feed downstream derivations.
```

Comment:
Feedback from Uwe:
> I did not get what this means - I would delete

Reply:
This sentence also appears in Recist 1.1 vignette. I think that in some studies not only soft tissue lesions are assessed, but also for example bone lesions, and then we also have data from radiological evaluation. Based on both types of data, we proceed further.

Reply:
Will remove until example data is included.